adapters/apex/apex.go

```diff
@@ -120,10 +116,19 @@ func New(options ...Option) (*Handler, error) {
 		}
 	}
 
-	// Run background scheduler.
-	var ctx context.Context
-	ctx, handler.cancel = context.WithCancel(context.Background())
-	go handler.run(ctx, handler.closeCh)
+	// Run background ingest.
+	go func() {
+		defer close(handler.closeCh)
+
+		res, err := handler.client.Datasets.IngestChannel(context.Background(), handler.datasetName, handler.eventCh, handler.ingestOptions...)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to ingest events: %s\n", err)
+		} else if res.Failed > 0 {
+			// Best effort on notifying the user about the ingest failure.
+			fmt.Fprintf(os.Stderr, "event at %s failed to ingest: %s\n",
+				res.Failures[0].Timestamp, res.Failures[0].Error)
+		}
```

Comment:
it's normally nice to have construction + running split up into 2 steps (ie this chunk would go in a Run() method where callers could decide whether or nor to use a goroutine). But I guess this is already public API that we can't change?

Reply:
You think so? I figure people want a seamless setup when setting up their logging: https://github.com/axiomhq/axiom-go/blob/main/examples/apex/main.go#L14-L17.

> _[...]_ where callers could decide whether or nor to use a goroutine _[...]_

Is there really a decision to be made? They need this running concurrently, no?

> But I guess this is already public API that we can't change?

We have yet to release a stable version. We can change everything.

Reply:
Makes sense. I'll do this in a follow up.